packages/datepicker/src/react/utils.ts

```diff
@@ -0,0 +1,244 @@
+      const firstTime = selected[0].getTime()
+      if (firstTime < dateTime) {
+        newDates.push(date)
+      } else {
+        newDates.unshift(date)
+      }
+      setSelected(newDates)
+    } else if (newDates.length === 2) {
+      setSelected([date])
+    }
+  } else {
+    newDates.push(date)
+    setSelected(newDates)
+  }
+  onSelect && onSelect(dateObj, evt)
+}
+
+const getDateIndex = (
+  selected: Date[],
+  condition: (time: number) => boolean
+) => {
+  let index
+  selected.some((date, i) => {
+    index = i
+    if (condition(date.getTime())) {
+      return true
+    }
+    // If we loop through all the selected dates and still didn't find
+    // one, make sure to add it to the end of the array.
+    index++
+    return false
+  })
+  return (index as unknown) as number
+}
+
+export const onMultiDateSelected = ({
+  selected = [],
+  setSelected,
+  onSelect
+}: OnMultipleDatesSelected) => (dateObj: DateObj, evt: SyntheticEvent) => {
+  const { selected: isSelected, selectable, date } = dateObj
+  if (!selectable) {
+    return
+  }
+  const newSelectedDates = selected.slice()
+  const selectedTime = date.getTime()
+  if (isSelected) {
+    const index = getDateIndex(selected, time => selectedTime === time)
+    newSelectedDates.splice(index, 1)
+  } else {
+    // Add
+    const index = getDateIndex(selected, time => selectedTime < time)
+    newSelectedDates.splice(index, 0, date)
+  }
+  setSelected(newSelectedDates)
+  onSelect && onSelect(dateObj, evt)
+}
+
+interface HandleChange<T> {
+  selected?: T
+  setSlide: React.Dispatch<React.SetStateAction<ValueOf<typeof slides>>>
+  setSelected: React.Dispatch<React.SetStateAction<T | undefined>>
+}
+
+export const useDateSelectChange = ({
+  selected,
+  setSlide,
+  setSelected
+}: HandleChange<Date>): [
+  string,
+  (event: React.ChangeEvent<HTMLInputElement>) => void
+] => {
+  const dateFormat = 'MM/dd/yyyy'
+  const [value, setValue] = React.useState<string>(
+    selected ? format(selected, dateFormat) : ''
+  )
+  useEffect(() => {
+    selected ? setValue(format(selected, dateFormat)) : ''
+  }, [selected])
+  const onChange: React.ChangeEventHandler<HTMLInputElement> = evt => {
+    const nextValue = evt.target.value
+    setValue(nextValue)
+    const fulldate = isMatch(nextValue, dateFormat)
+    try {
+      const nextSelected = fulldate
+        ? parse(nextValue, dateFormat, new Date())
+        : selected
+
+      if (
+        nextSelected instanceof Date &&
+        !isNaN((nextSelected as unknown) as number)
+      ) {
+        let nextSlide: ValueOf<typeof slides>
+        if (selected) {
+          nextSlide =
+            differenceInCalendarMonths(nextSelected, selected) > 0
+              ? 'forward'
+              : differenceInCalendarMonths(nextSelected, selected) < 0
+              ? 'backward'
+              : undefined
+        }
+        setSelected(nextSelected)
+        setSlide(nextSlide)
+      }
+    } catch (error) {
+      ;() => {}
```

Comment:
This is odd.  Why does this exist?

Reply:
Yeah this is no longer needed will remove